server/db/mysql/adapter.go

```diff
@@ -1250,8 +1334,14 @@ func (a *adapter) TopicsForUser(uid t.Uid, keepDeleted bool, opts *t.QueryOpt) (
 				break
 			}
 
+			// Optionally skip deleted topics.
+			if !keepDeleted && top.State == t.StateDeleted {
```

Comment:
Perhaps, modify the query to filter out deleted rows in sql?

Reply:
done